presto-orc/src/main/java/io/prestosql/orc/StripeReader.java

```diff
@@ -133,7 +133,7 @@ public Stripe readStripe(StripeInformation stripe, AggregatedMemoryContext syste
 
         // handle stripes with more than one row group
         boolean invalidCheckPoint = false;
-        if (stripe.getNumberOfRows() > rowsInRowGroup) {
+        if ((stripe.getNumberOfRows() > rowsInRowGroup) && rowsInRowGroup > 0) {
```

Comment:
The extra parentheses are not needed.  Also I would flip the `&&` and do the `> 0` first

Reply:
There are two other uses of `rowsInRowGroup` in this file, in the `createRowGroups` and `selectRowGroups` methods, which are only called from this `if` block.  Add `checkState` calls to both of those methods, so it is clear they should not be used in this case.

Reply:
Consider making `rowsInRowGroup` as `OptionalInt`, so it is super clear to future maintainers that this value may not be provided.

Reply:
Done. Handled all three comments.